File: system_lang/reports/combine_pdfs.pl
use strict;
use warnings;
use CAM::PDF;

# 結合するPDFファイルリスト
my @pdf_files = ('entrade.pdf', 'esist.pdf', 'zeta_dalanversian.pdf');
my $output_file = 'merged.pdf';

# PDFファイルを結合
my $pdf = CAM::PDF->new();
$pdf->addPage($_) for @pdf_files;

# ページ番号を変更
my $page_count = 1;
foreach my $file (@pdf_files) {
    my $pdf_part = CAM::PDF->new($file);
    my $num_pages = $pdf_part->numPages();
    
    for (my $i = 1; $i <= $num_pages; $i++) {
        my $page = $pdf_part->getPage($i);
        my $text = "Page $page_count";  # 新しいページ番号
        $page->{Contents} .= "\nBT /F1 12 Tf 100 700 Td ($text) Tj ET";  # ページ番号を追加
        $pdf->addPage($page);
        $page_count++;
    }
}

# 結合したPDFを保存
$pdf->clean();
$pdf->saveAs($output_file);

print "PDFファイルの結合とページ番号の書き換えが完了しました: $output_file\n";

File: system_lang/reports/combine_pdfs.rb
require 'combine_pdf'

# 結合するPDFファイルリスト
pdf_files = ['entrade.pdf', 'esist.pdf', 'zeta_dalanversian.pdf']
output_file = 'merged.pdf'

# PDFファイルを読み込み
combined_pdf = CombinePDF.new

pdf_files.each do |file|
  pdf = CombinePDF.load(file)
  combined_pdf << pdf
end

# ページ番号を追加
combined_pdf.pages.each_with_index do |page, index|
  page_number = index + 1
  page_text = "#{page_number}"
  
  # ページにテキストを追加
  page.text << page_text
  page.text_style = { size: 12, font: 'Helvetica' }
  page.text_position = [100, 700]  # テキストの位置を設定
end

# 結合したPDFを保存
combined_pdf.save(output_file)

puts "PDFファイルの結合とページ番号の書き換えが完了しました: #{output_file}"
